### PR TITLE
Karate Integration with ReportPortal

### PR DESCRIPTION
**Pull request by Shivanisingh05**
1. Added reportPortal properties in reportPortal.properties file
2. Added a class with Before and After methods, KarateReportPortalHook which implements interface RunTimeHook.
3. Added a class RPRepoter which has methods called in KarateReportPortalHook used to launch, send logs and results based on features/scenarios to report portal.
4. Used hook call in karate.run to initialise KarateReportPortalHook
 
**Below are the screenshots:**
1. Launch Dashboard
Screenshot 2023-07-10 at 3 30 31 PM

2. Feature Dashboard
Screenshot 2023-07-10 at 3 30 40 PM

3.Scenario Dashboard
Screenshot 2023-07-10 at 3 30 48 PM

4. Test Dashboard (With PASSED status)
Screenshot 2023-07-10 at 3 31 18 PM

5. Test Dashboard (With FAILED status)
Screenshot 2023-07-10 at 3 31 59 PM

### CODE REVIEW

File: src/test/java/com/znsio/reportPortal/KarateReportPortalHook.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        }
+        catch (Exception e)
+        {
+            logger.error("beforeSuite exception: {}", e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void afterSuite(Suite suite)
+    {
```
Comment (raghavgarg1996) [inline src/test/java/com/znsio/reportPortal/KarateReportPortalHook.java]: Code is not reformated

Comment (vineetjoc) [reply src/test/java/com/znsio/reportPortal/KarateReportPortalHook.java]: necessary changes are made in relevant before/after methods

File: src/test/resources/reportportal.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+rp.endpoint = http://localhost:8080
+rp.uuid = cb5a1d0b-c86e-4e2d-86cc-038dec764c52
+rp.launch = Getting-Started-With-Karate
+rp.project = superadmin_personal
+rp.username=default
+rp.enable=true
```
Comment (anandbagmar) [inline src/test/resources/reportportal.properties]: set default to false

File: src/test/java/com/znsio/reportPortal/RPReporter.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+            }
+        }
+        return false  ;
+    }
+
+    synchronized void startFeature(Feature feature) {
+        featureStartDate.put(getUri(feature), getTime());
+    }
+
+    synchronized void finishFeature(FeatureResult featureResult) {
```
Comment (anandbagmar) [inline src/test/java/com/znsio/reportPortal/RPReporter.java:101]: very complex method. break it up into smaller methods

File: src/test/resources/reportportal.properties (hunk 2 above)
Comment (Shivanisingh05) [reply src/test/resources/reportportal.properties]: rp.username is set to false

File: src/test/java/com/znsio/reportPortal/RPReporter.java (hunk 3 above)
Comment (Shivanisingh05) [reply src/test/java/com/znsio/reportPortal/RPReporter.java:101]: Method finishFeature has been broken down into smaller methods

File: src/test/resources/reportportal.properties (hunk 2 above)
Comment (anandbagmar) [reply src/test/resources/reportportal.properties]: I mean - set rp.enable=false by default

Comment (Shivanisingh05) [reply src/test/resources/reportportal.properties]: rp.enable is set to false

**Pull request merged**